analyses/new-feature-template.qmd

```diff
@@ -0,0 +1,1564 @@
+  rename_with(~ gsub("^meta_|^prox_|^other_|^loc_|^char_|^acs5|^acs_|^ccao_", "", .)) %>%
+  rename_with(~ gsub("_", " ", .)) %>%
+  cor() %>%
+  corrplot(method = "circle", tl.cex = 0.6, tl.srt = 45, addgrid.col = "grey", mar = c(1, 1, 1, 1))
+```
+
+# Overview of Model Metrics
+
+```{r model_stats}
+# Function to calculate the percentage difference
+percentage_diff <- function(x, y) {
+  scales::percent((y - x) / x)
+}
+
+
+min_n <- 10
+
+gte_n <- \(n_sales, min_n, fn, na_type) {
+  ifelse(sum(!is.na(n_sales)) >= min_n, fn, na_type)
+}
+rs_fns_list <- list(
+  cod_no_sop = \(x, y) gte_n(y, 2, cod(x / y, na.rm = TRUE), NA_real_),
+  prd_no_sop = \(x, y) gte_n(y, 2, prd(x, y, na.rm = TRUE), NA_real_),
+  prb_no_sop = \(x, y) gte_n(y, 2, prb(x, y, na.rm = TRUE), NA_real_),
+  mki_no_sop = \(x, y) gte_n(y, 2, mki(x, y, na.rm = TRUE), NA_real_),
+  cod = \(x, y) gte_n(y, min_n, cod(x / y, na.rm = TRUE), NA_real_),
+  cod_met = \(x, y) gte_n(y, min_n, cod_met(cod(x / y, na.rm = TRUE)), NA),
+  prd = \(x, y) gte_n(y, min_n, prd(x, y, na.rm = TRUE), NA_real_),
+  prd_met = \(x, y) gte_n(y, min_n, prd_met(prd(x, y, na.rm = TRUE)), NA),
+  prb = \(x, y) gte_n(y, min_n, prb(x, y, na.rm = TRUE), NA_real_),
+  prb_met = \(x, y) gte_n(y, min_n, prb_met(prb(x, y, na.rm = TRUE)), NA),
+  mki = \(x, y) gte_n(y, min_n, mki(x, y, na.rm = TRUE), NA_real_),
+  mki_met = \(x, y) gte_n(y, min_n, mki_met(mki(x, y, na.rm = TRUE)), NA)
+)
+
+
+
+ys_fns_list <- list(
+  rmse        = rmse_vec,
+  r_squared   = rsq_vec,
+  mae         = mae_vec,
+  mpe         = mpe_vec,
+  mape        = mape_vec,
+  mdape       = mdape_vec # From R/helpers.R
+)
+
+compute_stats <- function(data, group_var = NULL) {
+  if (!is.null(group_var)) {
+    data <- data %>%
+      group_by(across(all_of(group_var)))
+  }
+
+  result <- data %>%
+    summarize(
+      num_pin_no_class = n(),
+      num_sale_no_class = sum(!is.na(sale_ratio_study_price)),
+      rs_lst = list(map(rs_fns_list, ~ exec(.x, pmax(pred_pin_initial_fmv, 1), sale_ratio_study_price))),
+      ys_lst = list(map(ys_fns_list, ~ exec(.x, sale_ratio_study_price, pred_pin_initial_fmv))),
+      median_ratio = median(pred_pin_initial_fmv / sale_ratio_study_price, na.rm = TRUE),
+      .groups = ifelse(is.null(group_var), "drop", "keep")
+    ) %>%
+    unnest_wider(rs_lst) %>%
+    unnest_wider(ys_lst)
+
+  return(result)
+}
+
+df_stat_no_group <- compute_stats(assessment_pin)
+df_stat_triad <- compute_stats(assessment_pin, "meta_triad_code")
+df_stat_township <- compute_stats(assessment_pin, "meta_township_code")
+df_stat_nbhd <- compute_stats(assessment_pin, "meta_nbhd_code")
+
+df_stat_no_group_comparison <- compute_stats(assessment_pin_comparison)
+df_stat_triad_comparison <- compute_stats(assessment_pin_comparison, "meta_triad_code")
+df_stat_township_comparison <- compute_stats(assessment_pin_comparison, "meta_township_code")
+df_stat_nbhd_comparison <- compute_stats(assessment_pin_comparison, "meta_nbhd_code")
```

Comment:
**issue:** If you've run the model through the full pipeline and/or used Batch to run it, then these stats should already exist in the `model.performance` table. No need to recalculate them.

Reply:
Doesn't it only produce on the triad that it runs on?

Reply:
Edit again: When I run the query for S3, I get information for the target triad only. When I run a query like 
`select *
from model.performance
where model_run = 'whatever`
I get the correct information.